### PR TITLE
Clean up `DoubleLaunchChecker`

### PR DESCRIPTION
**Pull request by jglick**
In #6743 I noticed various nonstandard idioms in this class, which are cleaned up here. Switches the tool to use the regular administrative monitor system.

### Testing done

Added test coverage (using https://github.com/jenkinsci/jenkins-test-harness/pull/610). Set a breakpoint to interactively check that the GUI is displayed like other monitors and can be dismissed.

### Proposed changelog entries

- Switch the double-launch checker to a regular administrative monitor.

### Proposed upgrade guidelines

N/A

Before the changes are marked as `ready-for-merge`:

```[tasklist]
### Maintainer checklist
- [ ] There are at least two (2) approvals for the pull request and no outstanding requests for change.
- [ ] Conversations in the pull request are over, or it is explicit that a reviewer is not blocking the change.
- [ ] Changelog entries in the pull request title and/or **Proposed changelog entries** are accurate, human-readable, and in the imperative mood.
- [ ] Proper changelog labels are set so that the changelog can be generated automatically.
- [ ] If the change needs additional upgrade steps from users, the `upgrade-guide-needed` label is set and there is a **Proposed upgrade guidelines** section in the pull request title (see [example](https://github.com/jenkinsci/jenkins/pull/4387)).
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).
```

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/8127"></a>

### CODE REVIEW

File: test/pom.xml
Comment (jglick) [inline test/pom.xml:120]: (pending merge)

File: core/src/main/java/hudson/util/DoubleLaunchChecker.java
Comment (Vlatombe) [inline core/src/main/java/hudson/util/DoubleLaunchChecker.java:98]: Move the `isEnabled()` call above or in the `Schedule` class to avoid reading the file every time?

**Review by Vlatombe — APPROVED** [review]
Just a small suggestion, but looks good.

File: core/src/main/java/hudson/util/DoubleLaunchChecker.java
Comment (jglick) [reply core/src/main/java/hudson/util/DoubleLaunchChecker.java:98]: This is run only every 75m, and even then only mattering in the rare case that this monitor was activated and then disabled, and we are still _writing_ the same file unconditionally, so should not make much difference. Going to leave it as is to keep diff smaller.

**Review by timja — APPROVED** [review]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
